twitter/rate_limits.go

```diff
@@ -0,0 +1,60 @@
+package twitter
+
+import (
+	"net/http"
+
+	"github.com/dghubble/sling"
+)
+
+// RateLimitService provides methods for accessing Twitter rate limits
+// API endpoints.
+type RateLimitService struct {
+	sling *sling.Sling
+}
+
+// newRateLimitService returns a new RateLimitService.
+func newRateLimitService(sling *sling.Sling) *RateLimitService {
+	return &RateLimitService{
+		sling: sling.Path("application/"),
+	}
+}
+
+// RateLimitStatusResponse response of rate limit status
+type RateLimitStatusResponse struct {
+	RateLimitContext *RateLimitContext   `json:"rate_limit_context,omitempty"`
+	Resources        *RateLimitResources `json:"resources,omitempty"`
+}
+
+// RateLimitContext contains auth context
+type RateLimitContext struct {
+	AccessToken string `json:"access_token,omitempty"`
+}
+
+// RateLimitResources contains all limit status data for endpoints group by resources
+type RateLimitResources struct {
+	Users    map[string]*RateLimitResource `json:"users,omitempty"`
+	Statuses map[string]*RateLimitResource `json:"statuses,omitempty"`
+	Help     map[string]*RateLimitResource `json:"help,omitempty"`
+	Search   map[string]*RateLimitResource `json:"search,omitempty"`
```

Comment:
I see these are the 4 keys from the Twitter examples. Have you tried others to try to find the full set or resources?

Reply:
Looks like there are a lot more https://developer.twitter.com/en/docs/basics/rate-limits